### ISSUE TITLE
Question regarding sequences of animations

### ISSUE THREAD

**Issue by andreasbergqvist** [nandorojo/moti#51]
First, thanks for an awesome library!

I would like to so a sequence in an animation like:
```
<MotiView
  from={{ opacity: 0 }}
  animate={{
    opacity: [
      {
        value: 1,
        delay: 200,
        type: 'timing',
        duration: 1000,
      },
      {
        value: 0,
        delay: 4000,
        type: 'timing',
        duration: 500,
      },
    ],
  }} />
````
I.e, I would like to have sequenced animation of a value but with different durations.

Reading the documention at: https://moti.fyi/animations#sequence-animations
It sound like "Any transition settings can be passed to a sequence object."

But for me, the `duration` setting is not working (and is shown as faulty by TypeScript).

Is it possible somehow to sequence animations with different durations (and easings)?

**Comment by nandorojo** [comment]
Hey thanks for the issue. Yeah this is a known bug, the problem is at line 398 here: https://github.com/nandorojo/moti/blob/master/packages/core/src/use-map-animate-to-style.ts

I think it might be solved in Reanimated 2.1.0 if I use Object.assign rather than the ... spread there. You're welcome to take a look at that if you'd like, or I can try when I have some time.

**Comment by nandorojo** [comment]
Basically, if line 404 gets replaced with line 406, then the custom config should work. I just haven't had time to test if that works yet.

**Comment by andreasbergqvist** [comment]
Cool, I'll do some testing and get back!

**Comment by andreasbergqvist** [comment]
The duration (and easing) seem to work with the change.
But I'm having trouble getting the TypeScript typings working. Unsure if it is just my setup (I'm not used to Lerna) and how I'm building stuff or if there is an actual problems...

**Comment by nandorojo** [comment]
Could you elaborate?

**Comment by andreasbergqvist** [comment]
If I clone the repo, do the change and then using the examples/with-expo I'm getting TypeScript error for the duration property.
I'm testing `yarn build`. But I'm sure what is needed to use the examples folder for testing.

**Comment by nandorojo** [comment]
Are you setting `type: 'timing'`? The default is spring, which has no duration.

**Comment by nandorojo** [comment]
It's possible the example app has some messed up types, I'd have to look into that. But let me know if changing the type fixed it.

**Comment by andreasbergqvist** [comment]
Yeah, I've set type: 'timing'

**Comment by nandorojo** [comment]
Got it. So the runtime code works, but the types are breaking, right?

**Comment by andreasbergqvist** [comment]
Correct

**Comment by nandorojo** [comment]
Okay that's good. I'll take a look at the types when I have some time. Thanks for looking into this.

**Comment by nandorojo** [comment]
Looks like I figured out the types. I'll be pushing this shortly.

**Comment by nandorojo** [comment]
Released in `moti@0.9.0`!


https://user-images.githubusercontent.com/13172299/115479376-483efd80-a216-11eb-96be-f1f41a529c3d.mp4
